ui/src/app/machines/views/MachineDetails/MachineSummary/OverviewCard/MemoryCard/MemoryCard.tsx

```diff
@@ -0,0 +1,158 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+import { Button, Icon, ICONS, Tooltip } from "@canonical/react-components";
+
+import type { SetSelectedAction } from "../../MachineSummary";
+import { useSendAnalytics } from "app/base/hooks";
+import type { MachineDetails } from "app/store/machine/types";
+import { HardwareType } from "app/base/enum";
+
+type Props = {
+  machine: MachineDetails;
+  setSelectedAction: SetSelectedAction;
+};
+
+const hasTestsRun = (machine: MachineDetails, scriptType: string) => {
+  const testObj = machine[`${scriptType}_test_status`];
+  return (
+    testObj.passed + testObj.pending + testObj.running + testObj.failed > 0
+  );
+};
+
+const MemoryCard = ({ machine, setSelectedAction }: Props): JSX.Element => {
+  const sendAnalytics = useSendAnalytics();
+
+  const testsTabUrl = `/machine/${machine.system_id}/tests`;
+
+  return (
+    <>
+      <div className="overview-card__memory">
+        <strong className="p-muted-heading">Memory</strong>
+        <h4>{machine.memory ? machine.memory + " GiB" : "Unknown"}</h4>
+      </div>
+
+      <div className="overview-card__memory-tests u-flex--vertically">
+        <ul className="p-inline-list u-no-margin--bottom" data-test="tests">
+          {machine.memory_test_status.passed > 0 ? (
```

Comment:
Do you think this whole test traffic light thing could be extracted out into a component?

Reply:
Yep, I'll add a card for that